Breathing - Add cyanosis diagnose

MiszczuZPolski opened the pull request:
**When merged this pull request will:**
- Add cyanosise diagnose with 4 settable levels:
1. No Cyanosis - 100% SpO2 (no settable)
2. Slight Cyanosis - over 90% SpO2 (default)
3. Mild Cyanosis - over 75% SpO2 (default)
4. Severe Cyanosis - over 66% SpO2 (default)

Because some groups doesn't use lethal SpO2 value or have it different from default 65%, all levels are adjustable.

To do:

- Translation for more languages ._.

MiszczuZPolski committed:
First commit

MiszczuZPolski committed:
Log update

MiszczuZPolski committed:
Update stringtable.xml

MiszczuZPolski committed:
Update XEH_PREP.hpp

MiszczuZPolski committed:
fix

MiszczuZPolski committed:
fix

MiszczuZPolski committed:
ITS WORKING

MiszczuZPolski committed:
Message improvement

MiszczuZPolski committed:
adjusting SpO2 level to defaulf dieValue (65)

MiszczuZPolski committed:
Settings improvement

MiszczuZPolski committed:
little change

MiszczuZPolski committed:
Update stringtable.xml

bababoomjat commented:
The level of "no cyanosys" could be modified to be at least over ~95%? Or make it settable? 
I reckon that 100% is good for gameplay perspective, but from a immersion perspective a slight cianosis at 99% would mean that 3/4 of world population is cianotic.

https://journals.lww.com/pec-online/Abstract/2015/09000/Pulse_Oxygen_Saturation_Values_in_a_Healthy.7.aspx
This is a shoold-aged population but the concept is similar

And this without accounting for blood loss that could make cianosis even harder to detect, although i guess that such code would be much more difficult to implement

I don't mean to denigrate it, it's a great idea imo beside this detail!

MiszczuZPolski commented:
> The level of "no cyanosys" could be modified to be at least over ~95%? Or make it settable? I reckon that 100% is good for gameplay perspective, but from a immersion perspective a slight cianosis at 99% would mean that 3/4 of world population is cianotic.

I know IRL normal person would have SpO2 lower than 100%, but for gameplay purpose i want to keep it as 100% - that means when someone is injured (for example hidden pneumothorax) medic would know this. 

After commenting this i will add commit for medical level required so Anyone, Medic or Doctor could only detect it based on setting.

MiszczuZPolski committed:
Add medical level for action

MiszczuZPolski committed:
improvement

Added leftArm and rightArm to allowed selection
Moved log from activity to quick view

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
             <Japanese>SpO2(血中酸素濃度)が設定した値を下回ると気絶し、無意識状態に陥ります。</Japanese>
         </Key>
+        <Key ID="STR_kat_breathing_CheckCyanosis_Name">
+            <English>Check Cyanosis</English>
```
```suggestion
            <English>Check Cyanosis</English>
            <German>Zyanose überprüfen</German>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Sprawdź sinicę</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CheckCyanosis_Progress">
+            <English>Checking Cyanosis</English>
```
```suggestion
            <English>Checking Cyanosis</English>
            <German>Zyanose prüfen</German>
```

Tomcat-SG commented on addons/breathing/functions/fnc_treatmentAdvanced_CyanosisLocal.sqf:
```diff
@@ -0,0 +1,49 @@
+private _message = format ["%1",_messageCyanosis];
+[_message, 2, _player] call ace_common_fnc_displayTextStructured;
+
+[_target, "quick_view", LSTRING(CheckCyanosis_Log), [[_player] call ace_common_fnc_getName, _spO2Output]] call ace_medical_treatment_fnc_addToLog;
```
```suggestion
[_target, "quick_view", LSTRING(CheckCyanosis_Log), [_spO2Output]] call ace_medical_treatment_fnc_addToLog;
```

Tomcat-SG replied:
```suggestion
[_target, "quick_view", LSTRING(CheckCyanosis_Log), [_spO2Output]] call ace_medical_treatment_fnc_addToLog;
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Sprawdzanie sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CheckCyanosis_Log">
+            <English>Cyanosis Status: %2</English>
+            <Polish>Status sinicy: %2</Polish>
```
```suggestion
            <English>Cyanosis Status: %1</English>
            <German>Status der Zyanose: %1</German>
            <Polish>Status sinicy: %1</Polish>
```

Tomcat-SG replied:
```suggestion
            <English>Cyanosis Status: %1</English>
            <German>Status der Zyanose: %1</German>
            <Polish>Status sinicy: %1</Polish>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Status sinicy: %2</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CyanosisStatus_N">
+            <English>No Cyanosis</English>
```
```suggestion
            <English>No Cyanosis</English>
            <German>Keine Zyanose</German>
```

Tomcat-SG replied:
```suggestion
            <English>No Cyanosis</English>
            <German>Keine Zyanose</German>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Brak sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CyanosisStatus_Slight">
+            <English>Slight Cyanosis</English>
```
```suggestion
            <English>Slight Cyanosis</English>
            <German>Leichte Zyanose</German>
```

Tomcat-SG replied:
```suggestion
            <English>Slight Cyanosis</English>
            <German>Leichte Zyanose</German>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Lekka sinica</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CyanosisStatus_Mild">
+            <English>Mild Cyanosis</English>
```
```suggestion
            <English>Mild Cyanosis</English>
            <German>Mittelschwere Zyanose</German>
```

Tomcat-SG replied:
```suggestion
            <English>Mild Cyanosis</English>
            <German>Mittelschwere Zyanose</German>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Umiarkowana sinica</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CyanosisStatus_Severe">
+            <English>Severe Cyanosis</English>
```
```suggestion
            <English>Severe Cyanosis</English>
            <German>Schwere Zyanose</German>
```

Tomcat-SG replied:
```suggestion
            <English>Severe Cyanosis</English>
            <German>Schwere Zyanose</German>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Ciężka sinica</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_SETTING_Cyanosis">
+            <English>Enable cyanosis diagnose</English>
```
```suggestion
            <English>Enable cyanosis diagnose</English>
            <German>Zyanose-Diagnose aktivieren</German>
```

Tomcat-SG replied:
```suggestion
            <English>Enable cyanosis diagnose</English>
            <German>Zyanose-Diagnose aktivieren</German>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Włącz diagnozę sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_SETTING_Cyanosis_DESC">
+            <English>Enables cyanosis diagnose</English>
```
```suggestion
            <English>Enables cyanosis diagnose</English>
            <German>Aktiviert die Zyanose-Diagnose</German>
```

Tomcat-SG replied:
```suggestion
            <English>Enables cyanosis diagnose</English>
            <German>Aktiviert die Zyanose-Diagnose</German>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Włącza diagnozę sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_SETTING_slightValue">
+            <English>Slight cyanosis SpO2 value</English>
```
```suggestion
            <English>Slight cyanosis SpO2 value</English>
            <German>Leichter Zyanose SpO2-Wert</German>
```

Tomcat-SG replied:
```suggestion
            <English>Slight cyanosis SpO2 value</English>
            <German>Leichter Zyanose SpO2-Wert</German>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Poziom SpO2 dla lekkiej sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_SETTING_slightValue_DESC">
+            <English>Slight cyanosis value - default 90 SpO2</English>
```
```suggestion
            <English>Slight cyanosis value - default 90 SpO2</English>
            <German>Leichte Zyanose - Standardwert 90 SpO2</German>
```

Tomcat-SG replied:
```suggestion
            <English>Slight cyanosis value - default 90 SpO2</English>
            <German>Leichte Zyanose - Standardwert 90 SpO2</German>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Poziom SpO2 dla lekkiej sinicy - domyślnie 90 SpO2</Polish>
+        </Key>
+         <Key ID="STR_kat_breathing_SETTING_mildValue">
+            <English>Mild cyanosis SpO2 value</English>
```
```suggestion
            <English>Mild cyanosis SpO2 value</English>
            <German>Mittelschwere Zyanose SpO2-Wert</German>
```

Tomcat-SG replied:
```suggestion
            <English>Mild cyanosis SpO2 value</English>
            <German>Mittelschwere Zyanose SpO2-Wert</German>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Poziom SpO2 dla umiarkowanej sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_SETTING_mildValue_DESC">
+            <English>Mild cyanosis value - default 75 SpO2</English>
```
```suggestion
            <English>Mild cyanosis value - default 75 SpO2</English>
            <German>Mittelschwere Zyanose - Standardwert 75 SpO2</German>
```

Tomcat-SG replied:
```suggestion
            <English>Mild cyanosis value - default 75 SpO2</English>
            <German>Mittelschwere Zyanose - Standardwert 75 SpO2</German>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Poziom SpO2 dla umiarkowanej sinicy - domyślnie 75 SpO2</Polish>
+        </Key>
+         <Key ID="STR_kat_breathing_SETTING_severeValue">
+            <English>Severe cyanosis SpO2 value</English>
```
```suggestion
            <English>Severe cyanosis SpO2 value</English>
            <German>Schwere Zyanose SpO2-Wert</German>
```

Tomcat-SG replied:
```suggestion
            <English>Severe cyanosis SpO2 value</English>
            <German>Schwere Zyanose SpO2-Wert</German>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Poziom SpO2 dla ciężkiej sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_SETTING_severeValue_DESC">
+            <English>Severe cyanosis value - default 66 SpO2 \nShould be little bit higher than lethal SpO2 value</English>
```
```suggestion
            <English>Severe cyanosis value - default 66 SpO2 \nShould be little bit higher than lethal SpO2 value</English>
            <German>Schwere Zyanose - Standardwert 66 SpO2 \nSollte etwas höher sein als der tödliche SpO2-Wert</German>
```

Tomcat-SG replied:
```suggestion
            <English>Severe cyanosis value - default 66 SpO2 \nShould be little bit higher than lethal SpO2 value</English>
            <German>Schwere Zyanose - Standardwert 66 SpO2 \nSollte etwas höher sein als der tödliche SpO2-Wert</German>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Poziom SpO2 dla ciężkiej sinicy - domyślnie 66 SpO2 \nPowinien być minimalnie wyższy niż wartość śmiertelna SpO2</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CYANOSIS_TREATMENT_LEVEL">
+            <English>Check Cyanosis Minimum Level</English>
```
```suggestion
            <English>Check Cyanosis Minimum Level</English>
            <German>Zyanose prüfen Medic Level</German>
```

Tomcat-SG replied:
```suggestion
            <English>Check Cyanosis Minimum Level</English>
            <German>Zyanose prüfen Medic Level</German>
```

Tomcat-SG commented on addons/breathing/stringtable.xml:
```diff
@@ -752,5 +752,73 @@
+            <Polish>Minimalny poziom wyszkolenia do sprawdzenia sinicy</Polish>
+        </Key>
+        <Key ID="STR_kat_breathing_CYANOSIS_TREATMENT_LEVEL_DESCRIPTION">
+            <English>The minimum medical level required for someone to be able to check cyanosis.</English>
```
```suggestion
            <English>The minimum medical level required for someone to be able to check cyanosis.</English>
            <German>Das Medic Level, das erforderlich ist, damit jemand eine Zyanose feststellen kann.</German>
```

Tomcat-SG replied:
```suggestion
            <English>The minimum medical level required for someone to be able to check cyanosis.</English>
            <German>Das Medic Level, das erforderlich ist, damit jemand eine Zyanose feststellen kann.</German>
```

Tomcat-SG reviewed (changes requested):
German translation and argument for log changed

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/functions/fnc_treatmentAdvanced_CyanosisLocal.sqf

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG committed:
Update addons/breathing/stringtable.xml

Tomcat-SG reviewed (approved):
LGTM
